systemtests/test_end_to_end.py

```diff
@@ -0,0 +1,236 @@
+            os.mkdir(reduced_rb)
+            os.mkdir(reduced_auto)
+            os.mkdir(reduced_run)
+
+        def _find_run_in_database(self):
+            """
+            Find a ReductionRun record in the database
+            This includes a timeout to wait for several seconds to ensure the database has recieved
+            the record in question
+            :return: The resulting record
+            """
+            wait_times = [0, 2, 5, 10, 20]
+            results = []
+            for timeout in wait_times:
+                # Wait before attempting database access
+                print('Waiting for: {}'.format(timeout))
+                time.sleep(timeout)
+                # Check database has expected values
+                connection = self.database_client.get_connection()
+                results = connection.query(self.database_client.reduction_run()) \
+                    .join(self.database_client.reduction_run().instrument) \
+                    .join(self.database_client.reduction_run().status) \
+                    .join(self.database_client.reduction_run().experiment) \
+                    .filter(self.database_client.instrument().name == self.instrument) \
+                    .filter(self.database_client.reduction_run().run_number == self.run_number) \
+                    .all()
+                connection.commit()
+                try:
+                    actual = results[0]
+                except IndexError:
+                    # If no results found yet then continue
+                    continue
+                if actual.status.value == 'Completed' or actual.status.value == 'Error':
+                    print('Job reached {} status after {} seconds'.format(actual.status.value,
+                                                                          timeout))
+                    break
+            return results
+
+        @staticmethod
+        def _remove_run_from_database(instrument, run_number):
+            """
+            Uses the scripts.manual_operations.manual_remove script
+            to remove records added to the database
+            """
+            remove.remove(instrument, run_number)
+
+        @staticmethod
+        def _delete_reduction_directory():
+            """ Delete the temporary reduction directory"""
+            shutil.rmtree(os.path.join(get_project_root(), 'reduced-data'))
+
+        def tearDown(self):
+            """ Disconnect from services, stop external services and delete data archive """
+            self.queue_client.disconnect()
+            self.database_client.disconnect()
+            self._delete_reduction_directory()
+            del self.data_archive_creator
+            # Done in tearDown to ensure run is removed even if test fails early
+            self._remove_run_from_database(self.instrument, self.run_number)
+
+        @classmethod
+        def tearDownClass(cls):
+            # Stop external services
+            external.stop_activemq()
```

Comment:
Don't stop ActiveMQ when finished. I would assume that it is already running like the rest of the test suite and remove both start and stop. I would like to be able to run the entire test suite together. I can see that you have tried adding start and stop to other test classes, but I am still getting lots of connection refused errors on AMQ.

Reply:
I would argue that the better end solution is to start and stop activemq for all tests that require it as this ensures tests are run in an isolated manner and do not interplay with one another. However, given the growing size of this PR it's probably best to do this in a separate issue now. I'll remove it from this PR.

Reply:
That sounds like a good idea. The AMQ start/stop should be handled by a separate issue.